scripts/vllm_infer.py

```diff
@@ -137,31 +126,63 @@ def vllm_infer(
     else:
         lora_request = None
 
-    engine_args = {
-        "model": model_args.model_name_or_path,
-        "trust_remote_code": True,
-        "dtype": model_args.infer_dtype,
-        "max_model_len": cutoff_len + max_new_tokens,
-        "tensor_parallel_size": (get_device_count() // pipeline_parallel_size) or 1,
-        "pipeline_parallel_size": pipeline_parallel_size,
-        "disable_log_stats": True,
-        "enable_lora": model_args.adapter_name_or_path is not None,
-    }
-    if template_obj.mm_plugin.__class__.__name__ != "BasePlugin":
-        engine_args["limit_mm_per_prompt"] = {"image": 4, "video": 2, "audio": 2}
+    for i in range(0, len(train_dataset), batch_size):
+        print(f"Processing batch {i} to {min(i + batch_size, len(train_dataset))}...")
```

Comment:
```suggestion
for i in tqdm(range(0, len(train_dataset), batch_size), desc="Processing batches"):
```
we can use tqdm here

Reply:
That‘s a good idea!